be/src/agent/task_worker_pool.cpp

```diff
@@ -822,91 +822,118 @@ void* TaskWorkerPool::_publish_version_worker_thread_callback(void* arg_this) {
                     .build(&threadpool);
     assert(st.ok());
 
-    std::vector<TAgentTaskRequest> task_requests;
+    struct VersionCmp {
+        bool operator()(const std::unique_ptr<TAgentTaskRequest>& lhs,
+                        const std::unique_ptr<TAgentTaskRequest>& rhs) const {
+            // partition_version_infos order garanteed by FE
+            // so that we only compare first partition's version
+            for (auto& lhs_partition : lhs->publish_version_req.partition_version_infos) {
+                for (auto& rhs_partition : rhs->publish_version_req.partition_version_infos) {
+                    if (lhs_partition.partition_id == rhs_partition.partition_id &&
+                        lhs_partition.version > rhs_partition.version) {
+                        return true;
+                    }
+                    break;
+                }
+                break;
+            }
+            return false;
```

Comment:
Is this guarantee total order?  If lhs.partition_id != rhs.partition_id the result is always false, which means `lhs > rhs` and `rhs < rhs` can be both false

Reply:
just try best, there is an inconsistency between the partitions requested before and after, but the probability is low

Reply:
comparator should provide `strict weak ordering`, or this priority_queue won't work. 
https://en.cppreference.com/w/cpp/named_req/Compare

Reply:
why not just
`lhs->publish_version_req.partition_version_infos[0].verfsion > rhs->publish_version_req.partition_version_infos[0].verfsion`

Reply:
handle empty partition_version_infos